### PR TITLE
WIP: Index size records

### PR DESCRIPTION
**Pull request by stu-elastic**
Uses the `getLastCommittedSegmentInfos()` from `Engine` to periodically log segment records.

### CODE REVIEW

File: x-pack/plugin/metering-ix/src/main/java/org/elasticsearch/xpack/meter/ix/MeterIXPoller.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            for (final IndexShard shard : indexService) {
+                Engine engine = shard.getEngineOrNull();
+                if (engine == null) {
+                    continue;
+                }
+                long term = shard.getOperationPrimaryTerm();
+                int shardId = shard.shardId().id();
+                for (final SegmentCommitInfo commitInfo : engine.getLastCommittedSegmentInfos()) {
+                    try {
+                        var record = new IXRecord(
```
Comment (pgomulka) [inline x-pack/plugin/metering-ix/src/main/java/org/elasticsearch/xpack/meter/ix/MeterIXPoller.java:133]: do we need shardId, term etc or even index to report ix size?
when checking https://github.com/elastic/mx-team/blob/main/teams/billing/services/usage_record_schema_v2.md 
there isn't expectation for all the details?
as per that email in the past, perhaps even index is not necessary
https://groups.google.com/u/1/a/elastic.co/g/es-core-infra/c/8yNT08XlEPo?pli=1

File: x-pack/plugin/metering-ix/src/main/java/org/elasticsearch/xpack/meter/ix/MeterIXPoller.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import org.elasticsearch.indices.IndicesService;
+import org.elasticsearch.threadpool.Scheduler;
+import org.elasticsearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class MeterIXPoller implements LifecycleComponent {
```
Comment (thecoop) [inline x-pack/plugin/metering-ix/src/main/java/org/elasticsearch/xpack/meter/ix/MeterIXPoller.java:30]: AbstractLifecycleComponent does a lot of the logistics already?

File: x-pack/plugin/metering-ix/src/main/java/org/elasticsearch/xpack/meter/ix/MeterIXPoller.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                    } catch (IOException err) {
+                        logger.warn("error: [{}], index: [{}], shard: [{}], id: [{}]", err, index, shardId, commitInfo.getId());
+                    }
+                }
+            }
+        }
+        return records;
+    }
+
+    private record IXRecord(String index, int shardId, long term, long generation, String commitId, long size) {}
```
Comment (thecoop) [inline x-pack/plugin/metering-ix/src/main/java/org/elasticsearch/xpack/meter/ix/MeterIXPoller.java:152]: What's the term, generation, and commitId recorded for?

Comment (stu-elastic) [reply x-pack/plugin/metering-ix/src/main/java/org/elasticsearch/xpack/meter/ix/MeterIXPoller.java:152]: Removed in https://github.com/elastic/elasticsearch/pull/97978

**Comment by stu-elastic** [comment]
Superseded by https://github.com/elastic/elasticsearch/pull/97978

**Pull request closed without merging**